### PR TITLE
EPMRPP-33139. Added react-grid-layout to dashboard page

### PR DESCRIPTION
**Pull request by PavelZaletski**
Added react-grid-layout

### CODE REVIEW

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        return <div key={widgetId} className={cx('gadget-view')} data-grid={{ x, y, w, h }} />;
+      });
+    }
+
+    return (
+      <ScrollWrapper
+        autoHeight
+        autoHeightMax={this.props.isFullscreen ? window.screen.height : height}
+        hideTracksWhenNotNeeded
+        className={this.props.isFullscreen ? '' : 'no-scroll'}
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx]: Can this be achieved by using `renderTrackVertical` prop?

File: app/src/components/main/scrollWrapper/scrollWrapper.scss [hunk 2]
```diff
@@ -7,6 +7,12 @@
       opacity: 0.5;
     }
   }
+
+  &.no-scroll {
```
Comment (DiscoElevator) [inline app/src/components/main/scrollWrapper/scrollWrapper.scss]: Can this be achieved by using `renderTrackVertical` prop?

File: app/src/components/main/scrollWrapper/scrollWrapper.jsx [hunk 3]
```diff
@@ -73,10 +75,14 @@ export class ScrollWrapper extends Component {
...
+    const className = `${cx('scroll-component')} ${cx(this.props.className)}`;
+    return (
+      // base props are defined. For more info see https://github.com/malte-wessel/react-custom-scrollbars/blob/master/docs/API.md
       <Scrollbars
-        ref={(scrollbars) => { this.scrollbars = scrollbars; }}
-        className={cx('scroll-component')}
+        ref={(scrollbars) => {
+          this.scrollbars = scrollbars;
+        }}
+        className={className}
```
[the remark this answers is not in the document]
Comment (PavelZaletski) [reply app/src/components/main/scrollWrapper/scrollWrapper.jsx]: removed

File: app/src/pages/inside/dashboardItemPage/dashboardItemPage.jsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+          </div>
+        </div>
+
+        <Fullscreen enabled={this.state.isFull} onChange={this.onChangeFullscreen}>
+          <WidgetsGrid
+            dashboardId={this.props.match.params.dashboardId}
+            isFullscreen={this.state.isFull}
+          />
+          {this.state.isFull ? (
+            <i className={`${cx('icon-close')}`} onClick={this.closeFullscreen}>
```
[the remark this answers is not in the document]
Comment (PavelZaletski) [reply app/src/pages/inside/dashboardItemPage/dashboardItemPage.jsx]: fixed

File: app/src/pages/inside/dashboardItemPage/dashboardItemPage.jsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            <GhostButton icon={CancelIcon}>{formatMessage(messages.deleteWidget)}</GhostButton>
+          </div>
+        </div>
+
+        <Fullscreen enabled={this.state.isFull} onChange={this.onChangeFullscreen}>
+          <WidgetsGrid
+            dashboardId={this.props.match.params.dashboardId}
+            isFullscreen={this.state.isFull}
+          />
+          {this.state.isFull ? (
```
[the remark this answers is not in the document]
Comment (PavelZaletski) [reply app/src/pages/inside/dashboardItemPage/dashboardItemPage.jsx]: fixed

File: app/src/pages/inside/dashboardItemPage/dashboardItemPage.jsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    id: 'DashboardItemPage.fullscreen',
+    defaultMessage: 'Full screen',
+  },
+});
+
+@connect((state) => ({
+  url: `/api/v1/${activeProjectSelector(state)}/dashboard`,
+}))
+@injectIntl
+export class DashboardItemPage extends PureComponent {
```
[the remark this answers is not in the document]
Comment (PavelZaletski) [reply app/src/pages/inside/dashboardItemPage/dashboardItemPage.jsx]: fixed

File: app/src/pages/inside/dashboardItemPage/dashboardItemPage.scss [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  &:hover .icon-close {
+    opacity: 1;
+  }
+}
+
+.buttons-container {
+  height: 30px;
+  margin: 0;
+  padding: 20px 10px;
+  border-bottom: 1px solid #e9e9e9;
```
[the remark this answers is not in the document]
Comment (PavelZaletski) [reply app/src/pages/inside/dashboardItemPage/dashboardItemPage.scss]: fixed

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+      >
+        <ResponsiveGridLayout
+          rowHeight={rowHeight}
+          breakpoints={{ lg: 1200, md: 996, sm: 768, xs: 480, xxs: 0 }}
```
[the remark this answers is not in the document]
Comment (PavelZaletski) [reply app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx]: fixed

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+          onBreakpointChange={this.onBreakpointChange}
+          onDragStop={this.onGridChange}
+          onResizeStop={this.onGridChange}
+          cols={{ lg: 12, md: 12, sm: 1, xs: 1, xxs: 1 }}
```
[the remark this answers is not in the document]
Comment (PavelZaletski) [reply app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx]: fixed

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx (hunk 1 above)
Comment (PavelZaletski) [reply app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx]: removed

File: app/src/components/main/scrollWrapper/scrollWrapper.scss (hunk 2 above)
Comment (PavelZaletski) [reply app/src/components/main/scrollWrapper/scrollWrapper.scss]: removed

**Pull request closed without merging**